File: Max Level Sum in Binary Tree.cpp
class Solution{
  public:
    /*You are required to complete below method */
    int maxLevelSum(Node* root) {
        // Your code here
        if(root==NULL)
            return 0;
        queue<Node*> q;
        q.push(root);
        int su = root->data;
        while(!q.empty()){
            int tsum = 0;
            int sz = q.size();
            while(sz--){
                if(q.front()->left!=NULL)
                    q.push(q.front()->left);
                if(q.front()->right!=NULL)
                    q.push(q.front()->right);
                tsum += (q.front())->data;
                q.pop();
            }
            su = max(su, tsum);
        }
        return su;
    }
};

File: Max Level Sum in Binary Tree.java
 class Solution {
     public int maxLevelSum(Node root) {
        // add code here.
        if (root == null) {
            return 0;
        }

        Queue<Node> queue = new LinkedList<>();
        queue.add(root);
        int maxLevelSum = Integer.MIN_VALUE;

        while (!queue.isEmpty()) {
            int levelSum = 0;
            int levelSize = queue.size();

            for (int i = 0; i < levelSize; i++) {
                Node node = queue.remove();
                levelSum += node.data;

                if (node.left != null) {
                    queue.add(node.left);
                }

                if (node.right != null) {
                    queue.add(node.right);
                }
            }

            maxLevelSum = Math.max(maxLevelSum, levelSum);
        }

        return maxLevelSum;
    }
}
